### ISSUE TITLE
Need to reconnect when changing Ganache workspaces

### ISSUE THREAD

**Issue by wbt** [MetaMask/metamask-extension#8571]
**Describe the bug**
When doing some tests on a dapp recently, I noticed MetaMask performing super slow due to [Issue #6813](https://github.com/MetaMask/metamask-extension/issues/6813).  As a workaround, I got rid of the existing workspace in Ganache and replaced it with a new workspace, redeploying to the new workspace successfully.

However, in my dapp, I kept getting the error `Cannot read property '0' of null` when trying to do a `.call()` to read a value from a blockchain function.   I could not figure out where this was coming from.

Eventually, I found that switching MetaMask's networks selector away from the Ganache network (e.g. to mainnet) and back solved the issue.  I'm posting this here because it is a bug that wasted a nontrivial amount of time.  Ideally that bug will be fixed, but at least until then someone else encountering the issue might be able to find this page and the workaround above. 

**To Reproduce**
I have not tested minimal reproduction steps, but I'm guessing something like this:

1. Interact with a dapp running on local Ganache.
2. Discard that workspace and create a new one. 
3. Redeploy the app to the new workspace. 
4. Refresh the page in the browser window, and attempt to interact with the dapp.
5. Open the console to observe the errors, and notice the dapp not functioning correctly.

**Expected behavior**
MetaMask reconnects to the new Ganache workspace seamlessly, on page refresh.

**Versions:**
 - OS: Win10
 - Hardware Wallet: None
 - Browser: Google Chrome | 81.0.4044.138 (Official Build) (64-bit) (cohort: Stable)
 - MetaMask Version: 7.7.9
 - Ganache Version: 2.1.2

**Comment by jviray** [comment]
I'm having this exact same issue!

I noticed the error only happens when in the previous workspace/instance of ganache I deployed a contract and sent a transaction with Metamask (as opposed to just reading from the contract).

So if I've only called view functions from the contract, I can restart the workspace with no issue. Otherwise, I have to do the whole switching networks solution you talk about.

I was thinking it had something to do with the injected provider from Metamask not restarting or refreshing from the previous Ganache chain. Did you ever figure out what the exact cause? @wbt

**Comment by wbt** [comment]
@jviray No, I did not find an exact cause, especially once I found that workaround.  Your intuition makes sense though!

**Comment by jviray** [comment]
@wbt 

https://docs.metamask.io/guide/getting-started.html#basic-considerations

See paragraph 'Resetting Your Local Nonce Calculation'

> If you're running a test blockchain and restart it, you can accidentally confuse MetaMask because it calculates the next nonce based on both the network state and the known sent transactions.

Solution is to clear MetaMask's transaction queue, and effectively reset its nonce calculation. This is done through the Reset Account button in Settings --> Advanced --> Reset Account button (middle of list)

**Comment by wbt** [comment]
Hmm, that's odd. Usually the network toggle trick doesn't work for me when account reset is required.
